### PR TITLE
Added functionality to ignore .env # comments, trim EOL whitespace, detect different EOL terminators

### PR DESCRIPTION
**Pull request by seanblonien**
- Valid .env file comments that start with #, so added functionality to ignore anything after that symbol
  - Apparently [dontenv doesn't even have this feature](https://github.com/motdotla/dotenv/issues/524), but I figured couldn't hurt
- Also trimmed whitespace at end of line
- Changed it to auto-detect the EOL and split the string accordingly (allowing for \r\n and \n line endings)

### CODE REVIEW

File: lib/generateConfig.js [hunk 1]
```diff
@@ -26,7 +26,7 @@ async function generateConfig({ envConfig, envFile }) {
 
   const content = await readFileP(envFile, 'utf8');
 
-  content.split('\n').forEach((line) => {
+  content.split('\n').map(l => l.split('#')[0].trim()).forEach((line) => {
```
Comment (kHRISl33t) [inline lib/generateConfig.js]: Can you split these into multiple lines to make them more readable?

Something like:
```
content.split('\n')
   .map()
   .forEach()
```

**Review by kHRISl33t — CHANGES_REQUESTED** [review]
Nice touch to the package, good work, thank you! :)
One small comment about the style, otherwise it looks good.

**Comment by seanblonien** [comment]
If someone needs an environment variable value with a "#" in it, this solution doesn't really allow for that, but I think that's fine and not a common use-case, and easily solvable by replacing a pre-defined character pattern (such as HASH and doing a `window.__RUNTIME_CONFIG__.variableName.replace(/HASH/g, '#')` )  if someone truly needs that functionality

**Comment by seanblonien** [comment]
@kHRISl33t any more feedback on this?

**Comment by kHRISl33t** [comment]
@seanblonien will merge it later today (in a few hours), I just need to finish the docs for it :)
Sorry, I have been swamped with work in the last few days.

**Pull request merged**